File: src/EasyPrinter/Models/PrintSettings.cs
namespace EasyPrinter.Models
{
    /// <summary>
    /// Настройки печати
    /// </summary>
    public class PrintSettings
    {
        /// <summary>
        /// Количество копий (1-99)
        /// </summary>
        public int Copies { get; set; } = 1;

        /// <summary>
        /// Размер бумаги
        /// </summary>
        public PaperSize PaperSize { get; set; } = PaperSize.A4;

        /// <summary>
        /// Источник бумаги
        /// </summary>
        public PaperSource PaperSource { get; set; } = PaperSource.Auto;

        /// <summary>
        /// Качество печати
        /// </summary>
        public PrintQuality Quality { get; set; } = PrintQuality.Normal;

        /// <summary>
        /// Двусторонняя печать
        /// </summary>
        public DuplexMode Duplex { get; set; } = DuplexMode.None;

        /// <summary>
        /// Диапазон страниц (null = все страницы)
        /// </summary>
        public string? PageRange { get; set; }

        /// <summary>
        /// Страниц на листе
        /// </summary>
        public int PagesPerSheet { get; set; } = 1;

        /// <summary>
        /// Масштаб (25-400%)
        /// </summary>
        public int Scale { get; set; } = 100;

        /// <summary>
        /// Ориентация
        /// </summary>
        public PageOrientation Orientation { get; set; } = PageOrientation.Portrait;

        /// <summary>
        /// Настройки изображения
        /// </summary>
        public ImageAdjustments ImageAdjustments { get; set; } = new();
    }

    public enum PaperSize
    {
        A4,
        Letter,
        Legal,
        A5,
        Envelope10,
        EnvelopeC5,
        EnvelopeDL
    }

    public enum PaperSource
    {
        Auto,
        Tray1,
        ManualFeed
    }

    public enum PrintQuality
    {
        Draft,      // Черновик (600dpi)
        Normal,     // Нормальное
        High        // Высокое (FastRes 1200)
    }

    public enum DuplexMode
    {
        None,           // Односторонняя
        ManualDuplex    // Ручной дуплекс
    }

    public enum PageOrientation
    {
        Portrait,   // Книжная
        Landscape   // Альбомная
    }
}

File: easyprinter/models/print_settings.py
"""
Настройки печати
"""

from dataclasses import dataclass, field
from enum import Enum, auto
from typing import Optional

from .image_adjustments import ImageAdjustments


class PaperSize(Enum):
    """Размер бумаги"""
    A4 = auto()
    LETTER = auto()
    LEGAL = auto()
    A5 = auto()
    ENVELOPE_10 = auto()
    ENVELOPE_C5 = auto()
    ENVELOPE_DL = auto()


class PaperSource(Enum):
    """Источник бумаги"""
    AUTO = auto()
    TRAY1 = auto()
    MANUAL_FEED = auto()


class PrintQuality(Enum):
    """Качество печати"""
    DRAFT = auto()      # Черновик (600dpi)
    NORMAL = auto()     # Нормальное
    HIGH = auto()       # Высокое (FastRes 1200)


class DuplexMode(Enum):
    """Режим двусторонней печати"""
    NONE = auto()           # Односторонняя
    MANUAL_DUPLEX = auto()  # Ручной дуплекс


class PageOrientation(Enum):
    """Ориентация страницы"""
    PORTRAIT = auto()   # Книжная
    LANDSCAPE = auto()  # Альбомная


@dataclass
class PrintSettings:
    """Настройки печати"""

    # Количество копий (1-99)
    copies: int = 1

    # Размер бумаги
    paper_size: PaperSize = PaperSize.A4

    # Источник бумаги
    paper_source: PaperSource = PaperSource.AUTO

    # Качество печати
    quality: PrintQuality = PrintQuality.NORMAL

    # Двусторонняя печать
    duplex: DuplexMode = DuplexMode.NONE

    # Диапазон страниц (None = все страницы)
    page_range: Optional[str] = None

    # Страниц на листе
    pages_per_sheet: int = 1

    # Масштаб (25-400%)
    scale: int = 100

    # Ориентация
    orientation: PageOrientation = PageOrientation.PORTRAIT

    # Настройки изображения
    image_adjustments: ImageAdjustments = field(default_factory=ImageAdjustments)
